Add command to command palette and use SDK path

bombsimon opened the pull request:
After reading in #12 I started to look into how I could improve the error handling and feedback when we for example are missing a required launch option or if we cannot find specific tools. I also wanted to make use of plugin settings which seemed tricky. After reading some more on common capabilities and [commands](https://code.visualstudio.com/api/extension-capabilities/common-capabilities#command) (which is what the example plugin generates) I started to wonder if a debug session really was the right way to launch an external app.

I stumbled upon a few plugins that aim to launch external applications that are not debuggers
* [Android iOS Emulator](https://marketplace.visualstudio.com/items?itemName=DiemasMichiels.emulate)
* [Open in External App](https://marketplace.visualstudio.com/items?itemName=YuTengjing.open-in-external-app)
* [Open in Application](https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-open-in-application)
* [vscode-extension-samples](https://github.com/microsoft/vscode-extension-samples)

They all seem to be registring commands instead of extending the `(Logging)DebugSession`. When reading about [Debugger Extensions](https://code.visualstudio.com/api/extension-guides/debugger-extension) in the API docs it seems like it's not an obvious fit to just run two external commands (the `pdc` compiler and the simulator)

So, with all that said, what are your thoughts around converting this plugin to a regular VS Code command that would compile the code and run the simulator? It would make the users not have to setup either a task or a launch configuration and it would make it much easier to use the VS code API to show user feedback?

What do you think about a change like this?

**NOTE**: Like I mentioned before I'm quite new to VS Code extension coding and TS. If you think this is reasonable please give this a few test runs for your use-cases before merging to see if I'm doing anything stupid here.

**TODO**: I should probably update the screenshot in the `README` and includ the palette. 
![image](https://user-images.githubusercontent.com/3248948/162333232-92c2535c-e862-43b4-ab15-fcbc9b9083c0.png)

bombsimon commented on src/playdate.ts:
```diff
@@ -0,0 +1,80 @@
+
+    switch (process.platform) {
+    case "win32":
+        spawnSync("cmd", ["/c", pdc, `${ws}/source`, `${ws}/output.pdx`]);
```
Probably worth checking the exit code for all of these commands to show proper feedback if they fail.

bombsimon commented on README.md:
```diff
@@ -1,63 +1,33 @@
-1. Add a compile task by creating `.vscode/tasks.json`
+Either configure your environment variable `PLAYDATE_SDK_PATH` or add it in the
+settings. While in the settings it's also a good idea to add the Lua workspace
+libraries.:
```
```suggestion
libraries:
```

orta commented:
Can we do both? A launch task is conceptually what I always use to start a long running process - an command/action has always been something for a one-off

bombsimon commented:
> Can we do both? A launch task is conceptually what I always use to start a long running process - an command/action has always been something for a one-off

Yeah makes sense! I still find it hard to do proper feedback from the debug session but putting it back similar to how it was would make sense. I think I should be able to get the path to the SDK and launch it the same was from the debugger (to solve the issue from #12).

bombsimon commented:
@orta I pushed a few commits that cleans up even more and adds back support to launch `DebugSession`. However, I can't for my life get much things to work automatically from this debug session. In fact, I split up the `playdate.ts` to be able to call it from the `DebugSession` both with configuration and without but with no luck.

I ended up more or less duplicating the whole logic in the `DebugSession` but at least now I'm using the SDK path (which even that I can't seem to grab from either settings or the environment).

I'll try to see if I can find some more information and resources on how to hook in to the usual vscode API from the `DebugSession` but right now it doesn't seem to be very straight forward. Any help or pointers would be appreciated.

I haven't tried the latest revision on Windows and I'll be away without access to a Windows machine until next week.

bombsimon commented:
Added back debugger support and split code to be able to share how we launch from both command palette and debugger.

I've only tested this on macOS so far. I'll be away this week and won't have access to a Windows machine until next week to test.

PS. Sorry for squashing and force pushing but it got messy and hard to follow so I thought I'd clean up a bit before you got to a review.

orta commented on CHANGELOG.md:
```diff
@@ -1,9 +1,13 @@
-## [Unreleased]
+## [0.0.5]
+
+- add support for Winodws
```
```suggestion
- add support for Windows
```

orta commented:
Cool, yeah, looks good to me

orta commented:
Happy to make the release when you're ready

bombsimon commented on src/command.ts:
```diff
@@ -0,0 +1,69 @@
+
+  let result = playdate.compile(sdk, `${ws}/source`, `${ws}/output.pdx`);
+  if (result?.status !== 0 || !result) {
+    showMessage("failed to compile project");
```
```suggestion
    showMessage("Failed to compile project.");
```

bombsimon commented on src/command.ts:
```diff
@@ -0,0 +1,69 @@
+
+  let result = playdate.runSimulator(sdk, `${ws}/output.pdx`);
+  if (result?.status !== 0 || !result) {
+    showMessage("failed to start simulator");
```
```suggestion
    showMessage("Failed to start simulator.");
```

bombsimon commented on src/MockDebugSession.ts:
```diff
@@ -1,33 +1,26 @@
+    response: DebugProtocol.LaunchResponse,
+    args: LaunchRequestArguments
+  ) {
+    var sdkPath = args.sdkPath;
+    if (process.env.PLAYDATE_SDK_PATH) {
+      sdkPath = process.env.PLAYDATE_SDK_PATH;
```
Should this order be reversed, making the configuration JSON take precedence? 

```suggestion
    var sdkPath = process.env.PLAYDATE_SDK_PATH;
    if (args.sdkPath) {
      sdkPath = args.sdkPath;
```

orta replied:
Lets try it this way, and if someone wants to change it further down the line (I can see both making sense) then we can have that discussion

bombsimon reviewed (commented):
Tested on both Windows (11) and macOS now and after some minor fixes it now works on both platforms for me!

Appreciate if you want to give it a spin on any of your projects before merging to confirm this.
